src/products/utils/data.ts

```diff
@@ -276,3 +289,18 @@ export function mapFormsetStockToStockInput(
     warehouse: stock.id
   };
 }
+
+export const getEndPreorderDateInput = (
+  hasPreorderEndDate: boolean,
+  preorderEndDate?: string,
+  preorderEndHour?: string
+) =>
+  hasPreorderEndDate && preorderEndDate
+    ? `${preorderEndDate}T${preorderEndHour ?? "00:00"}`
+    : null;
+
+export const getPreorderEndDateFormData = (endDate?: string) =>
+  endDate ? endDate.split("T")?.[0] : null;
+
+export const getPreorderEndHourFormData = (endDate?: string) =>
+  endDate ? endDate.split("T")?.[1]?.split("+")?.[0] : null;
```

Comment:
You should be using native `Date` method (+ optionally already installed `moment` library) to handle Date parsing

Reply:
done